server/src/test/java/org/solarbank/server/unit/service/NasaClientTest.java

```diff
@@ -0,0 +1,118 @@
+package org.solarbank.server.unit.service;
+
+import java.io.IOException;
+import java.util.Map;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.solarbank.server.ErrorMessage;
+import org.solarbank.server.dto.CalculateRequest;
+import org.solarbank.server.dto.Location;
+import org.solarbank.server.service.NasaClient;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.solarbank.server.integration.IntegrationTestBase.createCalculateRequest;
+
+public class NasaClientTest {
```

Comment:
I really struggled to get unit testing to work for the WebClient, and there were lots of different suggestions on the internet. The below all work, but just not really sure about them as I use a real interpretation for the happy response but mocking for the unhappy response (I just couldn't manage to throw errors when using a real interpretation of the WebClient)